### ISSUE TITLE
Warning: Node of type rule not supported as an inline style

### ISSUE THREAD

**Issue by Echek99** [styled-components/styled-components#3741]
I am getting this warning while using StyledComponents with React Native, it seems that the problem comes when i use the CSS selector ::placeholder and i can't seem to find how to fix it.
`
const StyledInput = styled.TextInput`
  width: `100%;`
  min-width: 200px;
  height: 54px;
  padding: 10px;
  border-width: 1px;
  border-color: black;
  margin-bottom: 10px;
  &::placeholder{
    font-size: 20px;
    font-weight: bold;
    color: #34eb9e;
  }
`;
`

**Comment by joshbalfour** [comment]
You're right, it's because of the `&::placeholder`, you'll get the same message if you use any pseudo-selector.
You can't use pseudo-selectors with styled-components in react-native because styled-components converts the "css" you write to a style object.
If you're writing for web to, it's messy but you can wrap your pseudo-selector style with a check: 
```js
${Platform.OS === 'web' && css`
  &::placeholder{
    font-size: 20px;
    font-weight: bold;
    color: #34eb9e;
  }
`}
```

It's messy but gets rid of the error message.

**Comment by alireza-varmaghani** [comment]
I recived this Error in ract native and I releaized I had already created a subclass inside a StyledComponents. 
By removing that class there were no such a warning any more. 
Before:
```
const TestBox = styled(Box)`
           shadow-color: #fff;
           shadow-offset: {
            width: -1px;
            height: 1px;
          };
          shadow-opacity: 0.2;
          padding: 10px;`
```
After: 
```
 const TestBox = styled(Box)`
          shadow-color: #fff;
          shadow-opacity: 0.2;
          padding: 10px;`
```

**Comment by alirezajs** [comment]
I recived this Error in ract native and I releaized I had already created a subclass inside a StyledComponents. 
By removing that class there were no such a warning any more. 
Before:
```
const TestBox = styled(Box)`
           shadow-color: #fff;
           shadow-offset: {
            width: -1px;
            height: 1px;
          };
          shadow-opacity: 0.2;
          padding: 10px;`
```
After: 
```
 const TestBox = styled(Box)`
          shadow-color: #fff;
          shadow-opacity: 0.2;
          padding: 10px;`
```

**Comment by MarcosJBM** [comment]
> I recived this Error in ract native and I releaized I had already created a subclass inside a StyledComponents. By removing that class there were no such a warning any more. Before:
> 
> ```
> const TestBox = styled(Box)`
>            shadow-color: #fff;
>            shadow-offset: {
>             width: -1px;
>             height: 1px;
>           };
>           shadow-opacity: 0.2;
>           padding: 10px;`
> ```
> 
> After:
> 
> ```
>  const TestBox = styled(Box)`
>           shadow-color: #fff;
>           shadow-opacity: 0.2;
>           padding: 10px;`
> ```

This fixed my problem, even if I removed this from the code, the shadow still worked perfectly.

**Comment by quantizor** [comment]
Closing — this is a third-party library or toolchain compatibility issue, not a styled-components bug.
